examples/hosts/app-integration/external-controller/src/FocusTracker.ts

```diff
@@ -0,0 +1,134 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { EventEmitter } from "events";
+import {
+    DataObject,
+    DataObjectFactory,
+} from "@fluidframework/aqueduct";
+import { IEvent } from "@fluidframework/common-definitions";
+import { IMember, IServiceAudience } from "@fluid-experimental/fluid-static";
+import { IInboundSignalMessage } from "@fluidframework/runtime-definitions";
+
+export interface IFocusTrackerEvents extends IEvent {
+    (event: "focusChanged", listener: () => void): void;
+}
+
+/**
+ * Data object example of using the audience with signals to track focus
```

Comment:
i've heard people mention signals before, but not sure where I'd learn about them. Do we have any docs on them?

Reply:
i didn't find any docs; this was all done from code inspection and the existing usages (fortunately the api itself isn't too complicated, though i don't really know how it works under the hood).  it could definitely use some docs/better documentation in code